third_party/amd/lib/TritonAMDGPUTransforms/CanonicalizePointers.cpp

```diff
@@ -611,28 +611,71 @@ LogicalResult PointerCanonicalizer::rewriteForOp(scf::ForOp forOp,
                                                  Location curLoc,
                                                  OpOperand *curOperand,
                                                  Value &nextPtr) {
-  size_t operandNum = curOperand->getOperandNumber();
   FatPtr fatPtr = pointers[curOperand->get()];
   Value offset = fatPtr.offset;
   Value basePtr = fatPtr.basePtr;
 
-  // Replace the forOp with two additional argument (i.e., the curOperand's
-  // scalar pointer and the offset)
-  Value tensorPtr = createTensorPointer(fatPtr, curLoc);
-  auto newForOp =
-      replaceForOpWithNewSignature(rewriter, forOp, {basePtr, offset});
+  size_t operandNum = curOperand->getOperandNumber();
+  auto numLeadingInitsIncludingCurOp =
+      operandNum + 1 - forOp.getNumControlOperands();
+
+  SmallVector<Value> newInitArgs(
+      forOp.getInitArgs().take_front(numLeadingInitsIncludingCurOp));
```

Comment:
I would remove the currentOp. It was there before exactly because we were pushing back the new operands. We can now get rid of it (and make the IR way simpler).